### ISSUE TITLE
Attach: Add tests for attaching CLRIE to existing process

### ISSUE THREAD

**Issue by jander-msft** [microsoft/CLRInstrumentationEngine#208]
Adding some basic unit tests for attach should help with making sure that the attach feature doesn't regress. For example:
- Attach with a single instrumentation method should work.
- Attach with no instrumentation methods should either have CLRIE sit idle or reject profiling (need to define expected behavior).

**Comment by WilliamXieMSFT** [comment]
Add some more scenarios:
* Attaching CLRIE to a process that is currently profiled by CLRIE

**Comment by jander-msft** [comment]
@WilliamXieMSFT, not sure if that's something we really need to test. The CLR should be testing that already. The CLR is going to return CORPROF_E_PROFILER_ALREADY_ACTIVE if any profiler is already attached.

**Comment by WilliamXieMSFT** [comment]
@jander-msft good point. What about the scenario of the process is already profiled with Instrumentation Method 1 and we want to attach Instrumentation Method 2?

**Comment by jander-msft** [comment]
I think we're going to first get attach working with IMs already available upfront. Adding additional IMs after attach has completed will take a bit more investigation because we'll have to establish some communication model to allow for that information to be ingested into CLRIE, suspending the runtime while loading the new IMs, determining what event mask is allowed after attach has already completed, etc.
